### PR TITLE
docs: add publish endpoint

### PR DESCRIPTION
**Pull request by rogpeppe**
This just adds API docs for the publish endpoint without implementing it.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1591/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1592/
Test PASSed.

File: docs/API.md [hunk 1]
```diff
@@ -94,6 +93,24 @@ interested in public data only can include a `ignore-auth=1` query so that only
 public information is returned. In this case, results requiring authorization
 (if any) will be omitted.
 
+### Channels
+
+Any entity in the charm store is considered to be part of one or more "channels"
+(think "distribution channels"). Currently supported channels are "unpublished",
+"development" and "stable". All entities are initially (and always) part
+of the "unpublished" channel; subsequent operations on the publish
```
Comment (bac) [inline docs/API.md:101]: So if I upload a charm and then publish it, that charm is still in the "unpublished" channel?  Queries on the "unpublished" channel will return published charms?

Comment (mhilton) [inline docs/API.md:101]: Yes.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1593/
Test PASSed.

**Comment by frankban** [comment]
:+1:

File: docs/API.md (hunk 1 above)
Comment (bac) [inline docs/API.md:101]: That seems like it would surprise most users.

Comment (rogpeppe) [inline docs/API.md:101]: I agree. But these are the semantics that I have been asked to implement.

One possibility might be to rename "unpublished" to "initial".

**Comment by urosj** [comment]
:+1:

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**